src/github.com/getlantern/enproxy/proxy.go

```diff
@@ -128,6 +129,37 @@ func (p *Proxy) Serve(l net.Listener) error {
 	return httpServer.Serve(l)
 }
 
+func (p *Proxy) parseRequestProps(resp http.ResponseWriter, req *http.Request) (string, string, string, error) {
+	path := req.URL.Path
+
+	// If it's a reasonably long path, it likely follows our new request URI format:
+	// /X-Enproxy-Id/X-Enproxy-Dest-Addr/X-Enproxy-Op
+	if len(path) > 5 {
+		r, err := regexp.Compile("/(.*)/(.*)/(.*)")
```

Comment:
Hey yeah I was just fixing that too -- good catch! I did a little refactoring too.